policy/resource.go

```diff
@@ -44,8 +89,18 @@ func (r Resource) isObjectPattern() bool {
 
 // IsValid - checks whether Resource is valid or not.
 func (r Resource) IsValid() bool {
-	if strings.HasPrefix(r.Pattern, "/") {
-		return false
+	if r.isAWSS3() {
+		if strings.HasPrefix(r.Pattern, "/") {
+			return false
+		}
+	}
+	if r.isKMS() {
+		// TODO: Copied from KES repo, confirm that it is generally applicable.
+		if strings.IndexFunc(r.Pattern, func(c rune) bool {
+			return c == '/' || c == '\\' || c == '.'
```

Comment:
Is this a good rule for validating key names? Copied from KES.

Reply:
👍🏼